libcloud/storage/drivers/scaleway.py

```diff
@@ -0,0 +1,87 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from libcloud.storage.drivers.s3 import S3SignatureV4Connection, S3StorageDriver, S3_CDN_URL_EXPIRY_HOURS
+from libcloud.storage.drivers.s3 import BaseS3StorageDriver
+
+__all__ = ["ScalewayStorageDriver"]
+
+SCW_FR_PAR_STANDARD_HOST = "s3.fr-par.scw.cloud"
+SCW_NL_AMS_HOST = "s3.nl-ams.scw.cloud"
+SCW_PL_WAR_HOST = "s3.pl-waw.scw.cloud"
+
+# Maps SCW region name to connection hostname
+REGION_TO_HOST_MAP = {
+    "fr-par": SCW_FR_PAR_STANDARD_HOST,
+    "nl-ams": SCW_NL_AMS_HOST,
+    "pl-waw": SCW_PL_WAR_HOST,
+}
+
+NO_CDN_SUPPORT_ERROR = "CDN feature not implemented"
+
+
+class ScalewayStorageDriver(BaseS3StorageDriver):
+    name = "Scaleway Storage Driver"
+    website = "https://www.scaleway.com/en/object-storage/"
+    connectionCls = S3SignatureV4Connection
+    region_name = "fr-par"
+
+    def __init__(
+        self,
+        key,
+        secret=None,
+        secure=True,
+        host=None,
+        port=None,
+        region=None,
```

Comment:
Please also default ``region`` keyword argument to ``fr-par`` here.

Reply:
good catch.